### PR TITLE
FWI-1442 - Plugin for gathering newest image versions

### PR DESCRIPTION
**Pull request by jdesouza**
This PR fixes #

## Checklist
* [ ] I have signed the CLA
* [ ] I have updated/added any relevant documentation

### What alternative solution should we consider, if any?

[Internal Ticket FWI-1442](https://fairwinds.myjetbrains.com/youtrack/issue/FWI-1442)

### CODE REVIEW

File: plugins/trivy/cmd/trivy/main.go [hunk 1]
```diff
@@ -100,4 +118,43 @@ func main() {
...
+				})
+			}
+		}
+	}
+	return newImagesToScan
+}
+
+func getNewestVersions(versionsChan chan newestVersions, ctx context.Context, img models.ImageReport) {
+	repo := strings.Split(img.Name, ":")[0]
+	tag := strings.Split(img.Name, ":")[1]
```
Comment (rbren) [inline plugins/trivy/cmd/trivy/main.go:148]: Would be good to check that `img.Name` is the right format so this doesn't crash. E.g.

```suggestion
  parts := strings.Split(img.Name, ":")
  if len(parts) != 2 {
    return
  }
  repo := parts[0]
  tag := parts[1]
```

File: plugins/trivy/pkg/image/minimize.go [hunk 2]
```diff
@@ -24,14 +24,15 @@ func Minimize(images []models.ImageReport, lastReport models.MinimizedReport) mo
...
-			LastScan:       &timestamp,
+			ID:                 imageDetails.ID,
+			Name:               imageDetails.Name,
+			OwnerName:          imageDetails.OwnerName,
+			OwnerKind:          imageDetails.OwnerKind,
+			OwnerContainer:     imageDetails.OwnerContainer,
+			Namespace:          imageDetails.Namespace,
+			Report:             []models.VulnerabilityRefList{},
+			LastScan:           &timestamp,
+			RecommendationOnly: imageDetails.RecommendationOnly,
```
Comment (rbren) [inline plugins/trivy/pkg/image/minimize.go:35]: Before we merge this, we need to make sure the server is checking for this.

**Review by rbren — COMMENTED** [review]
This looks great!

File: plugins/trivy/cmd/trivy/main.go (hunk 1 above)
Comment (jdesouza) [reply plugins/trivy/cmd/trivy/main.go:148]: Good point

File: plugins/trivy/cmd/trivy/main.go [hunk 3]
```diff
@@ -100,4 +118,47 @@ func main() {
 	if err != nil {
 		panic(err)
 	}
+	logrus.Info("Finished writing file ", outputFile)
+}
+
+func getNewestVersionsToScan(ctx context.Context, imageWithVulns []models.ImageReport) []models.Image {
+	versionsChan := make(chan newestVersions, len(imageWithVulns))
+	for _, i := range imageWithVulns {
```
Comment (rbren) [inline plugins/trivy/cmd/trivy/main.go:126]: is it possible that there are duplicate images/repositories in here? we might be double-scanning if so

**Review by rbren — APPROVED** [review]
Thanks James!

LMK if it's possible to make this more efficient by only calling `getNewestVersions` once per repo, instead of once per image

**Pull request merged**